src/codeu/chat/server/Model.java

```diff
@@ -169,4 +228,178 @@ public void removeInterest(Uuid userId, Uuid interestId) {
+	} catch (FileNotFoundException e) {
+	  e.printStackTrace();
+	  return false;
+	}
+	
+	String completeText = "";
+	String line;
+	
+	// read the entirety of file into one String 
+	try {
+      while((line = br.readLine()) != null) {
+    	  completeText += line + "\n";
+      }
+      br.close();
+	} catch (IOException e) {
+	  e.printStackTrace();
+	  return false;
+	}
+	String[] jsonObjs = completeText.split("\n");
+	
+	if (jsonObjs.length == logSize) {
+	  // create gson object for deserializing
+	  GsonBuilder gb = new GsonBuilder();
+	  Type mapType = new TypeToken<Map<Uuid, UserType>>(){}.getType();
+	  gb.registerTypeAdapter(mapType, new PermissionAdaptor());
+	  Gson gson = gb.create();
+	  
+	  // convert from json to respective Store objects
+	  ArrayList<User> restoredUsers = gson.fromJson(jsonObjs[0].trim(), userType);
+	  ArrayList<ConversationHeader> restoredConvos = gson.fromJson(jsonObjs[1].trim(), conversationType);
+	  ArrayList<ConversationPermission> restoredPermissions = gson.fromJson(jsonObjs[2].trim(), permissionType);
+	  ArrayList<Message> restoredMessages = gson.fromJson(jsonObjs[3].trim(), messageType);
+	  ArrayList<Interest> restoredInterests = gson.fromJson(jsonObjs[4].trim(), interestType);
+	  ArrayList<ConversationPayload> restoredPayloads = gson.fromJson(jsonObjs[5].trim(), payloadType);
+	  
+	  // restore users
+	  for(int i = 0; i < restoredUsers.size(); i++) {
+	    add(restoredUsers.get(i));
+	  }
+	  
+	  // restore conversations
+	  for(int i = 0; i < restoredConvos.size(); i++) {
+	    add(restoredConvos.get(i));
+	  }
+	  
+	  // restore permissions
+	  for(int i = 0; i < restoredPermissions.size(); i++) {
+	    add(restoredPermissions.get(i));
+	  }
+	  
+	  // restore payloads
+	  for(int i = 0; i < restoredPayloads.size(); i++) {
+		add(restoredPayloads.get(i));
+	  }
+	  
+	  // restore messages
+	  for(int i = 0; i < restoredMessages.size(); i++) {
+	    add(restoredMessages.get(i));
+	  }
+	  
+	  // restore interests
+	  for(int i = 0; i < restoredInterests.size(); i++) {
+		Interest temp = restoredInterests.get(i);
+		addInterest(temp.id, temp.userId, temp.interestId, temp.type, temp.lastUpdate);
+	  }
+	  
+	  System.out.println("Sucessfully restored previous state of the Server");
+	  
+	  return true;
+	} else {
+	  System.out.println("ERROR: incorrect number of elements in log");
+	  System.out.println("Expected: " + logSize + "\t Actual: " + jsonObjs.length);
+	  return false;
+	}
+  }
+  
+  public String createFilePath() {
+	String workingDirectory = System.getProperty("user.dir");
+	return workingDirectory + File.separator + "serverLog.txt";
+  }
+  
+  public class PermissionAdaptor extends TypeAdapter<Map<Uuid,UserType>> {
+	public Map<Uuid,UserType> read(JsonReader reader) throws IOException {
+	  if (reader.peek() == JsonToken.NULL) {
+	    reader.nextNull();
+	    return null;
+	  }
+	  String hashMap = reader.nextString();
```

Comment:
In general, I would prefer not to do your own serialization, and always rely on serialization libraries if available.

If we didn't have a hard deadline (which is actually pretty rare in real life), I would ask you to spend time fixing it. Here, I'll approve, since I trust you and trust the group as a whole to triage any remaining issues with the time you have left.